projecttacoma/fqm-testify
# Implement exporting of all test cases

# Summary
Adds a button and logic for downloading all test cases to disk as a zip file. All test cases take the form of a patient bundle.

## New Behavior
This addresses the following user story: “As a user, I want to be able to export and download a test case that I have created so I can share it with my team.”

This PR specifically allows the user to download all the test cases with a single button click.

## Code Changes
- Brings in [JSZip](https://stuk.github.io/jszip/) and [FileSaver](https://www.npmjs.com/package/file-saver) as dependencies for generating a zip and saving it to disk
- New function in `ResourceCreationPanel` component to construct a bundle string for all the current patients, create a file for each of them, and compile those files into a zip
- New function in `downloadUtil` to use `FileSaver` to create a Blob from the JSZip object and save it to disk
- New button in `ResourceCreationPanel` component that will kick off the download

# Testing guidance
* Run the unit tests with `npm run test`.
* Run `npm run dev` and navigate to `localhost:3000` (or whatever port you are testing on).
* Try creating multiple patients, adding resources to all of them, and then clicking the "Download All Patients" button. Confirm that the downloaded zip file contains appropriate patient bundles for each of the created test cases.
* Try creating one patient and check that the zip file only contains one patient bundle.
* Check that the button for downloading all the patients is disabled unless we have at least one test case already created.

(I had some trouble unit testing the actual file download with `FileSaver`, so suggestions would be appreciated!)
